pkgs/development/haskell-modules/with-packages-wrapper.nix

```diff
@@ -46,14 +47,18 @@ let
   llvm          = lib.makeBinPath
                   ([ llvmPackages.llvm ]
                    ++ lib.optional stdenv.isDarwin llvmPackages.clang);
+  ghcLibdirLink = runCommand "ghc_libdir" {} ''
+    mkdir -p '${libDir}'
+    echo '${ghcLibdir}' > '${libDir}/ghc_libdir_override'
```

Comment:
@cstrahan, I am not aware of any particular reason why we shouldn't be able to switch to `symlinkJoin`. I assume that the `lndir` tool on which that function is based is no less portable that our implementation of `buildEnv` is ... so IMHO we should use whichever one is more convenient. Generally speaking, being able to rely on `bin` and `etc` referring to real directories in a ghcWithPackages environment sounds like a good idea.